starlette/background.py

```diff
@@ -39,5 +41,7 @@ def add_task(
         self.tasks.append(task)
 
     async def __call__(self) -> None:
-        for task in self.tasks:
-            await task()
+        async with anyio.create_task_group():
+            with anyio.CancelScope(shield=True):
+                for task in self.tasks:
```

Comment:
Could we make this opt in on a per-task basis? For backwards compatibility and because I don't think in some cases it _does_ make sense to cancel the background task if the client closes the connection.

Reply:
Which cases?

Reply:
Business logic. Like if you're doing something that is expensive and you don't care if it succeeds or fails if the client closes the connection. No _specific_ use case in mind, just thinking in general.

Although now I realize this is even more complicated because the behavior changes depending on if there is a `BaseHTTPMiddleware` involved, right? What happens if there isn't a `BaseHTTPMiddleaware` involved? Does it depend on the ASGI server?

Reply:
> fails if the client closes the connection

bkg tasks runs after the response 🤔